### DISCUSSION TITLE
Combine screw up multipage TIFFs?

### DISCUSSION THREAD

**Discussion by vortah** [ImageMagick/ImageMagick#3021] [Help]
Is it me or -combine screws up multipage TIFF pages?

Example:

magick -depth 8 -colorspace sRGB -type truecolor -verbose 1.pdf \
-channel R -combine -define colorspace:auto-grayscale=off -alpha off -compress lzw -verbose 1.tiff

... and all pages are mixed into one image. What im doing wrong, considering I just want to grab a specific channel, not losing multi page option in resulting TIFF?

> jake@iMac Downloads % magick -version
Version: ImageMagick 7.0.10-52 Q16 x86_64 2020-12-20 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5) 
Delegates (built-in): bzlib freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib

**Comment by snibgo** [comment]
Why are you using "-combine"? What do you want it to do?

**Reply by vortah** [reply]
Now when you asked.. The whole idea was to convert PDF -> TIFF with these specs: RED channel = 8 bit, GREEN = 1 bit, BLUE = 1 bit, making it multi page just like a source. When I started browsing docs and reading forums, this led me to combine as it gave me closest to was was needed result. For example, identify shows:

>   Filename: 1.tiff
>   Format: TIFF (Tagged Image File Format)
>   Mime type: image/tiff
>   Class: DirectClass
>   Geometry: 595x842+0+0
>   Resolution: 72x72
>   Print size: 8.26389x11.6944
>   Units: Undefined
>   Colorspace: sRGB
>   Type: Palette
>   Base type: TrueColor
>   Endianness: LSB
>   Depth: 8-bit
>   Channel depth:
>     Red: 8-bit
>     Green: 1-bit
>     Blue: 1-bit
>

.. no multipage, though.

Honestly speaking, I'm not 100% sure what combine does, it's just the closest I could get to what I was looking for at first place:(

**Comment by vortah** [comment]
Found another method to lower green/blue channels to 1 bit, while keeping red at 8bit.

magick -colorspace sRGB -type truecolor -verbose 1.pdf \
-define colorspace:auto-grayscale=off -compress lzw -depth 8 -channel green,blue -fx '0' -verbose 1.tiff

While it worked with rose, it didn't with the PDF-TIFF conversion for some reason - multipage conversion is gone leaving in output first page only.

**Comment by snibgo** [comment]
You haven't said what you are trying to do.

If you want to make a multi-image sRGB TIFF from a multi-page PDF, then:
```
magick -colorspace sRGB -type truecolor -verbose 1.pdf 1.tiff
```
It doesn't matter if the image needs 8 bits for red and only 1 bit for green and blue.

If that command doesn't do what you need, then please say what you do need, and give a link to your PDF.

**Reply by vortah** [reply]
I had to begin with what I am trying to achieve at first place. My fault, sorry

I am trying to convert a PDF into multipage TIFF with these **exact** specs as in example

> Image:
>   Filename: example.tiff
>   Format: TIFF (Tagged Image File Format)
>   Mime type: image/tiff
>   Class: PseudoClass
>   Geometry: 2479x3508+0+0
>   Resolution: 300x300
>   Print size: 8.26333x11.6933
>   Units: PixelsPerInch
>   Colorspace: sRGB
>   Type: Palette
>   Base type: TrueColor
>   Endianness: LSB
>   Depth: 8-bit
>   Channel depth:
>     Red: 8-bit
>     Green: 1-bit
>     Blue: 1-bit
> 

Example: [download link](https://dropmefiles.com/384034)

**Comment by snibgo** [comment]
You have linked to a TIFF file with multiple images. The images are different, with different requirements for bit depth. For example, the final image has only two colours, black and white, so needs only 1 bit for each RGB channel.

**Reply by vortah** [reply]
You mean I am heading towards wrong direction? Then how do I specify bit depth to be 8 and colorspace (truecolor) for the 1st page and make other pages extracted from PDF to be 1-bit, if it's possible at all?

**Comment by snibgo** [comment]
IM will use whatever bit depth is necessary. For example, to see the bit depth of all channels in all images:
```
magick result.tiff -verbose info: |grep bit

  Depth: 8-bit
    Red: 8-bit
    Green: 8-bit
    Blue: 8-bit
  Depth: 8-bit
    Red: 1-bit
    Green: 1-bit
    Blue: 1-bit
  Depth: 8-bit
    Red: 1-bit
    Green: 1-bit
    Blue: 1-bit
  Depth: 8-bit
    Red: 1-bit
    Green: 1-bit
    Blue: 1-bit
  Depth: 8-bit
    Red: 1-bit
    Green: 1-bit
    Blue: 1-bit
  Depth: 8-bit
    Red: 1-bit
    Green: 1-bit
    Blue: 1-bit
  Depth: 8-bit
    Red: 1-bit
    Green: 1-bit
    Blue: 1-bit
```
For just the first image in the tiff, there are only 15 different colours, and IM uses a palette to store the image. The palette has 255 entries, but only 15 entries are used.

**Reply by vortah** [reply]
Thanks for the insights! These two days since your reply I was trying to get an identical output to a 8-bit image with these particular specs:

>   Filename: 1page.tiff
>   Format: TIFF (Tagged Image File Format)
>   Mime type: image/tiff
>   Class: PseudoClass
>   Geometry: 2479x3508+0+0
>   Resolution: 300x300
>   Print size: 8.26333x11.6933
>   Units: PixelsPerInch
>   Colorspace: sRGB
>   Type: Palette
>   Endianness: LSB
>   Depth: 8-bit
>   Channel depth:
>     Red: 8-bit
>     Green: 8-bit
>     Blue: 8-bit

the command is:
magick -density 300 -units PixelsPerInch -type palette '1.pdf[0]' -alpha off -define colorspace:auto-grayscale=off -define tiff:rows-per-strip=3508 -depth 8 -compress lzw -verbose 1.tiff  

  Filename: 1.tiff
  Format: TIFF (Tagged Image File Format)
  Mime type: image/tiff
  Class: PseudoClass
  Geometry: 2479x3508+0+0
  Resolution: 300x300
  Print size: 8.26333x11.6933
  Units: PixelsPerInch
  Colorspace: sRGB
  **Type: Grayscale
  Base type: Palette**
  Endianness: LSB
  **Depth: 4/8-bit**
  Channel depth:
    Red: 8-bit
    Green: 8-bit
    Blue: 8-bit

If -type palette in the command is replaced with -type truecolor, I am getting DirectClass instead of PseudoClass and same Greyscale type I cannot get rid of >_< Though, the depth has changed and isn't mixed anymore.

>   Filename: 1.tiff
>   Format: TIFF (Tagged Image File Format)
>   Mime type: image/tiff
>   Class: DirectClass
>   Geometry: 2479x3508+0+0
>   Resolution: 300x300
>   Print size: 8.26333x11.6933
>   Units: PixelsPerInch
>   Colorspace: sRGB
>   Type: Grayscale
>   Base type: TrueColor
>   Endianness: LSB
>   Depth: 8-bit
>   Channel depth:
>     Red: 8-bit
>     Green: 8-bit
>     Blue: 8-bit

How do I get rid of -Type: Grayscale and make it Palette instead without mixed channels and such exactly as in the first example? What am I missing?

**Comment by snibgo** [comment]
"magick identify -verbose x.tiff" or "magick x.tiff -verbose info:" gives information about the image, after IM has read the image from the file.

"Type:" examines all the pixels (or, for palette images, all the palette entries), and describes how they might be summarised, the _potential_ type.
```
Type: Grayscale
```
This means that all the pixels in the image are gray.

```
Type: Palette
```
This means that not all the pixels are gray, and there are 256 or fewer unique colours.

```
Type: Grayscale
Base type: Palette
```
If the image in the file has a type that is different to the _potential_ type, the file type is given as the "Base type". In this example, the file type is palette, and IM has examined the palette entries, and they are all gray.

```
Depth: 4/8-bit
```
This means the file had 4 bits per channel per pixel. IM doesn't store 4-bit numbers, so has expanded this to 8 bits.

> How do I get rid of -Type: Grayscale and make it Palette instead...?

If all the pixels are gray, the "Type" will always be Grayscale. If you don't want "Type: Grayscale" then ensure that at least one pixel is not gray.

Why does this matter to you? I suspect "Type:" doesn't matter to you at all. Perhaps you really care about how the image is stored in the file. Perhaps exiftool would be more useful for that.

**Comment by fmw42** [comment]
Please post your original PDF file.

**Comment by vortah** [comment]
Thanks to everyone for answers, saved me days, literally.

Last question, is it possible to change image Depth without changing channel-specific depth? 

```
  Colorspace: sRGB
  Type: Bilevel
  Base type: Palette
  Endianness: LSB
  Depth: 8-bit
  Channel depth:
    Red: 1-bit
    Green: 1-bit
    Blue: 1-bit

```

I've got 1-bit everywhere, want to make it look like in an example shown above

**Comment by snibgo** [comment]
My answer would repeat my previous post.

But I will add that your image is bilevel, which means it has only two colours: black and white. What does "image depth" matter when you have only black and white?
